torch/distributed/rpc/api.py

```diff
@@ -592,9 +594,15 @@ def remote(to, func, args=None, kwargs=None, timeout=UNSET_RPC_TIMEOUT):
                 **kwargs,
             )
         else:
-            (pickled_python_udf, tensors) = _default_pickler.serialize(
-                PythonUDF(func, args, kwargs)
-            )
+            arg_cache_key = (func, args, str(kwargs))
```

Comment:
Would this cache_key work if args or kwargs themselves contain an unhashable type like dict?

Reply:
Good point, this wouldn't work if args contains a dict since it would try to hash that. We might need to use `str(args)` as part of the key as well, performance permitting.